Content.Server/Antag/MobReplacementRuleSystem.cs

```diff
@@ -1,37 +1,172 @@
+    [Dependency] private readonly ChatSystem _chatSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+    [Dependency] private readonly IComponentFactory _componentFactory = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly NpcFactionSystem _npcFaction = default!;
+    [Dependency] private readonly NPCSystem _npc = default!;
+    [Dependency] private readonly TransformSystem _transform = default!;
+    [Dependency] private readonly AdvertiseSystem _advertise = default!;
 
     protected override void Started(EntityUid uid, MobReplacementRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
         var query = AllEntityQuery<VendingMachineComponent, TransformComponent>();
         var spawns = new List<(EntityUid Entity, EntityCoordinates Coordinates)>();
+        var stations = _gameTicker.GetSpawnableStations();
 
         while (query.MoveNext(out var vendingUid, out _, out var xform))
         {
-            if (!_random.Prob(component.Chance))
+            var ownerStation = _stationSystem.GetOwningStation(vendingUid);
+
+            if (ownerStation == null
+                || ownerStation != stations[0])
+                return;
+
+            // Make sure that we aren't running this on something that is already a mimic
+            if (HasComp<CombatModeComponent>(vendingUid))
                 continue;
 
             spawns.Add((vendingUid, xform.Coordinates));
         }
+        if (spawns == null)
+        {
+            //WTF:THE STATION DOESN'T EXIST, WE MUST BE IN A CONTENT.INTEGRATION TEST! QUICK, PUT A MIMIC AT 0,0!!!
+            Spawn(component.Proto, new EntityCoordinates(uid, new Vector2(0, 0)));
+        }
+        else
+        {
+            //This is intentionally not clamped. If a server host wants to replace every vending machine in the entire station with a mimic, who am I to stop them?
+            var k = MathF.MaxMagnitude(component.NumberToReplace, 1);
+            while (k > 0 && spawns != null)
+            {
+                var spawnLocation = _random.PickAndTake(spawns);
+                BuildAMimicWorkshop(spawnLocation.Entity, component);
+
+                if (k == MathF.MaxMagnitude(component.NumberToReplace, 1)
+                    && component.DoAnnouncement)
+                    _chatSystem.DispatchStationAnnouncement(stations[0], Loc.GetString("station-event-rampant-intelligence-announcement"), playDefaultSound: true,
+                        colorOverride: Color.Red, sender: "Central Command");
+
+                k--;
+            }
+        }
+    }
+
+    /// <summary>
+    /// It's like Build a Bear, but MURDER
+    /// </summary>
+    /// <param name="uid"></param>
+    public void BuildAMimicWorkshop(EntityUid uid, MobReplacementRuleComponent component)
```

Comment:
```suggestion
    public void BuildAMimic(EntityUid uid, MobReplacementRuleComponent component)
```

Reply:
No.

Reply:
Why?

Reply:
I wanted to leave a silly joke in the code. I mean if it must be removed I'll remove it. :)